plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -549,15 +576,50 @@ static obs_properties_t *vaapi_properties(void *unused)
 				       obs_module_text("VAAPI.Device"),
 				       OBS_COMBO_TYPE_LIST,
 				       OBS_COMBO_FORMAT_STRING);
-	char path[32] = "/dev/dri/renderD1";
-	for (int i = 28;; i++) {
-		sprintf(path, "/dev/dri/renderD1%d", i);
-		if (access(path, F_OK) == 0) {
-			char card[128] = "Card: ";
-			sprintf(card, "Card%d: %s", i - 28, path);
-			obs_property_list_add_string(list, card, path);
-		} else {
-			break;
+	if (os_file_exists("/dev/dri/by-path")) {
+		os_dir_t *by_path_dir = os_opendir("/dev/dri/by-path");
+		struct pci_access *pacc;
+		struct os_dirent *file;
+		char pci_slot[13], type[7], *card_name;
+		int len;
+
+		pacc = pci_alloc();
+		pci_init(pacc);
+		while ((file = os_readdir(by_path_dir)) != NULL) {
+			// file_name pattern: pci-<pci_slot::12>-<type::{"card","render"}>
+			char *file_name = file->d_name;
+			if (strcmp(file_name, ".") == 0 ||
+			    strcmp(file_name, "..") == 0)
+				continue;
+			len = strlen(file_name);
+			strncpy(type, file_name + 17, len - 16);
+			if (strcmp(type, "render") == 0) {
+				strncpy(pci_slot, file_name + 4, 12);
+				pci_slot[12] = 0;
+				card_name = get_device_name_from_pci(pacc,
+								     pci_slot);
+				if (card_name == NULL)
+					continue;
+				char path[64] = "/dev/dri/by-path/";
```

Comment:
When omitting curly braces, insert a blank line between the content that would have been enclosed and the following content.